src/frontend/src/optimizer/plan_node/generic/scan.rs

```diff
@@ -343,6 +344,9 @@ impl GenericPlanNode for Scan {
     }
 
     fn stream_key(&self) -> Option<Vec<usize>> {
+        if matches!(self.table_catalog.table_type, TableType::Internal) {
+            return None;
+        }
         let id_to_op_idx = Self::get_id_to_op_idx_mapping(&self.output_col_idx, &self.table_desc);
         self.table_desc
             .stream_key
```

Comment:
Doesn't the issue mean that the `stream_key` field of `TableDesc` is not reasonable?

Reply:
It still sounds strange that `TableType::Table` has `stream_key` even we handled `TableType::Internal`...